Basic usage in Typescript

I'm trying to use the package in a typescript project. I attempted to use the code described in the Basic Usage section in a .ts file verbatim:

Users.ts
```
import { MongoDataSource } from 'apollo-datasource-mongodb'

export default class Users extends MongoDataSource {
  getUser(userId) {
    return this.findOneById(userId)
  }
} 
```
and later on import it and use it as a data source:

index.ts
```
import Users from './data-sources/Users.js'
...
const server = new ApolloServer({
  ...
  dataSources: () => ({
    users: new Users(client.db().collection('users'))
  })
})
```

But I am getting errors that keep the code from compiling.

- Users.ts my editor (vscode) reports: `Generic type 'MongoDataSource<TData, TContext>' requires between 1 and 2 type arguments.ts(2707)` on MongooseDataSource when I am extending the class.
- index.ts: my editor reports `Expected 0 arguments, but got 1.ts(2554)` on `client.db().collection('users')`
- On compiling, get: 
```
../src/index.ts:434
    return new TSError(diagnosticText, diagnosticCodes)
           ^
TSError: ⨯ Unable to compile TypeScript:
src/index.ts:34:7 - error TS2322: Type '() => { users: Users; }' is not assignable to type '() => DataSources<object>'.
  Type '{ users: Users; }' is not assignable to type 'DataSources<object>'.
    Property 'users' is incompatible with index signature.
      Type 'Users' has no properties in common with type 'DataSource<object>'.

34       dataSources: () => ({
         ~~~~~~~~~~~

  node_modules/apollo-server-core/dist/graphqlOptions.d.ts:24:5
    24     dataSources?: () => DataSources<TContext>;
           ~~~~~~~~~~~
    The expected type comes from property 'dataSources' which is declared here on type 'ApolloServerExpressConfig & { cors?: boolean | CorsOptions | undefined; onHealthCheck?: ((req: Request<ParamsDictionary, any, any, ParsedQs>) => Promise<...>) | undefined; }'
src/index.ts:35:31 - error TS2554: Expected 0 arguments, but got 1.

35         users: new Users(client.db().collection("users")),
```

I didn't go farther in the example code because of these issues. 

Are there any examples with TypeScript code for the basic usage or could you point me in the right direction on how to fix the issues I am having?

Thanks in advance.

--- a/README.md
+++ b/README.md
@@ -20,6 +20,7 @@ This package uses [DataLoader](https://github.com/graphql/dataloader) for batchi
   - [Basic](#basic)
   - [Batching](#batching)
   - [Caching](#caching)
+  - [TypeScript](#typescript)
 - [API](#api)
   - [findOneById](#findonebyid)
   - [findManyByIds](#findmanybyids)
@@ -164,6 +165,58 @@ const resolvers = {
 ```
 
 Here we also call [`deleteFromCacheById()`](#deletefromcachebyid) to remove the user from the cache when the user's data changes. If we're okay with people receiving out-of-date data for the duration of our `ttl`—in this case, for as long as a minute—then we don't need to bother adding calls to `deleteFromCacheById()`.
+
+### TypeScript
+
+Since we are using a typed language, we want the provided methods to be correctly typed as well. This requires us to make the `MongoDataSource` class polymorphic. It requires 1-2 template arguments. The first argument is the type of the document in our collection. The second argument is the type of context in our GraphQL server, which defaults to `any`. For example:
+
+`data-sources/Users.ts`
+
+```ts
+import { MongoDataSource } from 'apollo-datasource-mongodb'
+import { ObjectId } from 'mongodb'
+
+interface UserDocument {
+  _id: ObjectId
+  username: string
+  password: string
+  email: string
+}
+
+// This is optional
+interface Context {
+  loggedInUser: UserDocument
+}
+
+export default class Users extends MongoDataSource<UserDocument, Context> {
+  getUser(userId) {
+    // this.context has type `Context` as defined above
+    // this.findOneById has type `(id: ObjectId) => Promise<UserDocument | null | undefined>`
+    return this.findOneById(userId)
+  }
+}
+```
+
+and:
+
+```ts
+import { MongoClient } from 'mongodb'
+
+import Users from './data-sources/Users.ts'
+
+const client = new MongoClient('mongodb://localhost:27017/test')
+client.connect()
+
+const server = new ApolloServer({
+  typeDefs,
+  resolvers,
+  dataSources: () => ({
+    users: new Users(client.db().collection('users'))
+    // OR
+    // users: new Users(UserModel)
+  })
+})
+```
 
 ## API
 

